portal/src/graphql/portal/AuthenticationLoginIDSettings.tsx

```diff
@@ -0,0 +1,599 @@
+
+  const {
+    value: reservedKeywords,
+    onChange: onReservedKeywordsChange,
+  } = useTextField(reservedKeywordsConfig.join(", "));
+  const {
+    value: blockedDomains,
+    onChange: onBlockedDomainsChange,
+  } = useTextField(blockedDomainsConfig.join(", "));
+  const {
+    value: isBlockReservedKeywords,
+    onChange: onIsBlockReservedKeywordsChange,
+  } = useCheckbox(reservedKeywordsConfig.length > 0);
+  const {
+    value: isBlockDomains,
+    onChange: onIsBlockDomainsChange,
+  } = useCheckbox(blockedDomainsConfig.length > 0);
+  const {
+    value: isIncludeFreeEmailDomains,
+    onChange: onIsIncludeFreeEmailDomainsChange,
+  } = useCheckbox(false);
+  const {
+    value: isEmailCaseSensitive,
+    onChange: onIsEmailCaseSensitiveChange,
+  } = useCheckbox(!!emailConfig?.case_sensitive);
+  const {
+    value: isIgnoreDotLocal,
+    onChange: onIsIgnoreDotLocalChange,
+  } = useCheckbox(!!emailConfig?.ignore_dot_sign);
+  const { value: isAllowPlus, onChange: onIsAllowPlusChange } = useCheckbox(
+    !emailConfig?.block_plus_sign
+  );
+
+  const onSaveButtonClicked = React.useCallback(() => {
+    if (props.appConfig == null) {
+      return;
+    }
+
+    const validationData = {
+      reservedUsername,
+      excludedKeywords,
+      reservedKeywords,
+      blockedDomains,
+    };
+
+    const { errorResult, isValid } = validateInput(
+      validationData,
+      validationRules
+    );
+    setErrorState(errorResult);
+    if (!isValid) {
+      return;
+    }
+
+    const screenState = {
+      usernameEnabled,
+      emailEnabled,
+      phoneNumberEnabled,
+
+      reservedUsername,
+      excludedKeywords,
+      isBlockReservedUsername,
+      isIncludeDefaultReservedUsernameList,
+      isExcludeKeywords,
+      isIncludeDefaultKeywordList,
+      isUsernameCaseSensitive,
+      isAsciiOnly,
+
+      reservedKeywords,
+      blockedDomains,
+      isBlockReservedKeywords,
+      isBlockDomains,
+      isIncludeFreeEmailDomains,
+      isEmailCaseSensitive,
+      isIgnoreDotLocal,
+      isAllowPlus,
+    };
+
+    const newAppConfig = constructAppConfigFromState(
+      props.appConfig,
+      screenState
+    );
+    // TODO: call mutation to save config
+    console.log(newAppConfig);
+  }, [
+    props.appConfig,
+
+    usernameEnabled,
+    emailEnabled,
+    phoneNumberEnabled,
+
+    reservedUsername,
+    excludedKeywords,
+    isBlockReservedUsername,
+    isIncludeDefaultReservedUsernameList,
+    isExcludeKeywords,
+    isIncludeDefaultKeywordList,
+    isUsernameCaseSensitive,
+    isAsciiOnly,
+
+    reservedKeywords,
+    blockedDomains,
+    isBlockReservedKeywords,
+    isBlockDomains,
+    isIncludeFreeEmailDomains,
+    isEmailCaseSensitive,
+    isIgnoreDotLocal,
+    isAllowPlus,
+  ]);
+
+  return (
+    <div className={styles.root}>
+      <div className={styles.widgetContainer}>
+        <ExtendableWidget
+          extendable={usernameEnabled}
```

Comment:
I'm not sure if we should not let user toggle the panel when the feature is disabled. I think we can still let user see the content of panel while the feature is disabled.